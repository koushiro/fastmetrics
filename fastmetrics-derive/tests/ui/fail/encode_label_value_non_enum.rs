use fastmetrics_derive::EncodeLabelValue;

// This should fail because EncodeLabelValue is designed for enums
#[derive(EncodeLabelValue)]
struct NotAnEnum {
    field: String,
}

fn main() {}
