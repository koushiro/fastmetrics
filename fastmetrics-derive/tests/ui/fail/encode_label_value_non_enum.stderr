error: #[derive(EncodeLabelValue)] can only be used on enums
 --> tests/ui/fail/encode_label_value_non_enum.rs:5:8
  |
5 | struct NotAnEnum {
  |        ^^^^^^^^^
